src/Parser.y

```diff
@@ -222,15 +216,24 @@ smac	:: { (AlexPosn,String) }
  	: '.'				{ (tokPosn $1, ".") }
 	| SMAC				{ case $1 of T p (SMacT s) -> (p, s) }
 
+char	:: { Char }
+	: DIGIT				{ $1 }
+	| CHAR				{ $1 }
+
+natnum	:: { Int }
+	: digit				{ $1 }
+	| natnum digit			{ $2 * 10 + $1 }
```

Comment:
Should be `$1 * 10 + $2`.
Is this what breaks #141: https://github.com/simonmar/alex/runs/4911840593?check_suite_focus=true#step:22:116
Anyway, fixed this in #201.

Reply:
Ahh yes, thanks.

Reply:
Oh, I realized another problem with recognizing numbers in the parser.  `natnum` happily accepts digit separated by spaces, like `1 4` in `r{1 4,1 4}` is happily accepted as 14 repetitions of `r` now.
I do not think we want to allow that.
How about a middle ground, recognizing numbers in the scanner, but not storing them as `Integer`, but as `String`, so we can get them back into character sequences. I'll play with this solution in #199 a bit...

Reply:
Maybe we can use lexer states to only lex numerals inside the multiplicity-brackets `{nnn,mmm}`.

Reply:
Good point, and both of those look interesting to me.